pyserini/index/pyutils.py

```diff
@@ -35,13 +35,105 @@ class IndexUtils:
     '''
 
     def __init__(self, index_dir):
-        self.object = JIndexUtils(JString(index_dir))
+        self.object = JIndexReaderUtils()
+        self.reader = self.object.getReader(JString(index_dir))
 
     class DocumentVectorWeight:
         NONE = JDocumentVectorWeight.NONE
         TF_IDF = JDocumentVectorWeight.TF_IDF
 
-    def dump_document_vectors(self, reqDocidsPath: str, weight: DocumentVectorWeight):
+    class Posting:
+        '''
+        Basic Posting class for Postings List
+        '''
+        def __init__(self, docid, term_freq, positions):
+            self.docid = docid
+            self.term_freq = term_freq
+            self.positions = positions
+
+        def __repr__(self):
+            repr = '(' + str(self.docid) + ', ' + str(self.term_freq) + ')'
+            if self.positions:
+                repr += ' [' + ','.join([str(p) for p in self.positions]) + ']'
+            return repr
+
+    def analyze_term(self, term):
+        '''
+        Parameters
+        ----------
+        term : str
+            Search term
+        Returns
+        -------
+        result : str
+            Stemmed term
+        '''
+        return self.object.analyzeTerm(self.reader, JString(term))
+
+    def get_term_counts(self, term):
+        '''
+        Parameters
+        ----------
+        term : str
+            Search term
```

Comment:
note that this is before analysis/stemming?

Reply:
True, I just followed the order in Anserini. Should I change both then?

Reply:
sorry, wasn't clear - I meant, "term prior to stemming/analysis"

Reply:
Ahh sorry, my bad... Fixed.